common/screenshotter.go

```diff
@@ -175,27 +175,38 @@ func (s *screenshotter) screenshot(
 
 	// Add clip region
 	//nolint:dogsled
-	_, visualViewport, _, _, _, _, err := cdppage.GetLayoutMetrics().Do(cdp.WithExecutor(s.ctx, sess))
+	_, _, _, _, visualViewport, _, err := cdppage.GetLayoutMetrics().Do(cdp.WithExecutor(s.ctx, sess))
 	if err != nil {
 		return nil, fmt.Errorf("getting layout metrics for screenshot: %w", err)
 	}
 
+	visualViewportScale := 1.0
+	visualViewportPageX, visualViewportPageY := 0.0, 0.0
+	// we had a null pointer panic cases, when visualViewport is nil
+	// instead of the erroring out, we fallback to defaults and still try to do a screenshot
+	// ideally this case also should be logged
+	if visualViewport != nil {
```

Comment:
Can we log an error when `visualViewport` and/or `cssVisualViewport` are `nil`? It would be good to track this in case we are able to replicate it in the future.

Reply:
Yeah, sure. Like I wrote in the comment it's good to log the case and details that help us to identify the root case. It's just screenshoter has no logger, but I can inject it there 🤝

Reply:
Done with the latest fp